lib/Chunk.js

```diff
@@ -185,7 +186,11 @@ class Chunk {
 	hasRuntime() {
 		for (const chunkGroup of this._groups) {
 			// We only need to check the first one
-			return chunkGroup.isInitial() && chunkGroup.getRuntimeChunk() === this;
+			return (
+				chunkGroup.isInitial() &&
+				// prettier-ignore
+				/** @type {Entrypoint} */ (chunkGroup).getRuntimeChunk() === this
```

Comment:
Add a `instanceof Entrypoint` check instead of casting.

Reply:
Alternatively, if Entrypoint.isInitial is equivalent to `instanceof Entrypoint`, then changing the return type of Entrypoint.isInitial to `@returns {this is Entrypoint}` would accomplish the same thing.

Reply:
Using `Instanceof` will require importing `Entrypoint` which is not optimal. If that's okay, is this what you're suggesting?

```js
return (
    chunkGroup.isInitial() &&
    chunkGroup instanceof Entrypoint &&
    chunkGroup.getRuntimeChunk() === this
);
```

@sandersn per my experience `this is Type` does not work with JSDoc.

Reply:
Just did `chunkGroup instanceof Entrypoint`